assets/queries/k8s/seccomp_profile_is_not_configured/query.rego

```diff
@@ -1,199 +1,91 @@
 	}
 }
 
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	metadata := document.spec.jobTemplate.spec.template.metadata
-
-	parentMetadata := document.metadata
-	common_lib.valid_key(metadata, "annotations")
-	annotations := metadata.annotations
+# pod defines seccompProfile.type and container inherits this setting
+checkSeccompProfile(specInfo, container, containerType, document, metadata) = result {
+	containerCtx := object.get(container.securityContext, "seccompProfile", {})
+	not common_lib.valid_key(containerCtx, "type")
 
-	common_lib.valid_key(annotations, "seccomp.security.alpha.kubernetes.io/defaultProfileName")
-
-	seccomp := annotations["seccomp.security.alpha.kubernetes.io/defaultProfileName"]
-
-	seccomp != "runtime/default"
+	profile := specInfo.spec.securityContext.seccompProfile.type
+	not any([profile == "RuntimeDefault", profile == "Localhost"])
 
 	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name={{%s}}.spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName", [parentMetadata.name]),
+		"documentId": document.id,
+		"searchKey": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type=%s", [metadata.name, specInfo.path, profile]),
 		"issueType": "IncorrectValue",
-		"keyExpectedValue": "'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is 'runtime/default'",
-		"keyActualValue": sprintf("'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is '%s'", [seccomp]),
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type should be set to 'RuntimeDefault' or 'Localhost'", [metadata.name, specInfo.path]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.securityContext.seccompProfile.type is set to '%s'", [metadata.name, specInfo.path, profile]),
 	}
 }
 
-###################################################################
+# neither pod nor container define seccompProfile.type
+checkSeccompProfile(specInfo, container, containerType, document, metadata) = result {
+	specCtx := object.get(specInfo.spec, "securityContext", {})
+	specSeccompCtx := object.get(specCtx, "seccompProfile", {})
+	not common_lib.valid_key(specSeccompCtx, "type")
 
-#general
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") != "Pod"
-	object.get(document, "kind", "undefined") != "CronJob"
-
-	metadata := document.spec.template.metadata
-
-	parentMetadata := document.metadata
-	not common_lib.valid_key(metadata, "annotations")
+	common_lib.valid_key(container, "securityContext")
+	containerSeccompCtx := object.get(container.securityContext, "seccompProfile", {})
+	not common_lib.valid_key(containerSeccompCtx, "type")
```

Comment:
Therefore, the desired condition here is: `type` is not defined in a _potentially_ existing `container.securityContext.seccompProfile` object. 

Please note that the line you suggested would evaluate to _false_, in case `seccompProfile` is not yet existing in `container.securityContext`. In practice, `seccompProfile` has no effect unless specified with a concrete type. 

The check `common_lib.valid_key(container, "securityContext")` was added for situations where deprecated pod annotations are used. In other words, the rule deliberately evaluates to _false_, if no securityContext is defined. This is usually the case for legacy manifests using classic PodSecurityPolicies together with pod annotations. The _negative1.yaml_ testcase is one example where we would see an _MissingAttribute_ alert otherwise.

`containerSeccompCtx := object.get(container.securityContext, "seccompProfile", {})` allows `seccompProfile` to _potentially_ exist.